libraries/api-extractor-model/src/mixins/ApiItemContainerMixin.ts

```diff
@@ -209,6 +274,173 @@ export function ApiItemContainerMixin<TBaseClass extends IApiItemConstructor>(
+          // We add the to-be-added members to an intermediate array instead of immediately
+          // to the maps themselves to support method overloads with the same name.
+          if (ApiNameMixin.isBaseClassOf(member)) {
+            if (!membersByName.has(member.name)) {
+              membersToAdd.push(member);
+            }
+          } else {
+            if (!membersByKind.has(member.kind)) {
+              membersToAdd.push(member);
+            }
+          }
+        }
+
+        for (const member of membersToAdd) {
+          if (ApiNameMixin.isBaseClassOf(member)) {
+            const members: ApiItem[] = membersByName.get(member.name) || [];
+            members.push(member);
+            membersByName.set(member.name, members);
+          } else {
+            const members: ApiItem[] = membersByKind.get(member.kind) || [];
+            members.push(member);
+            membersByKind.set(member.kind, members);
+          }
+        }
+
+        // Interfaces can extend multiple interfaces, so iterate through all of them.
+        const extendedItems: ApiItem[] = [];
+        let extendsTypes: readonly HeritageType[] | undefined;
+
+        switch (next.kind) {
+          case ApiItemKind.Class: {
+            const apiClass: ApiClass = next as ApiClass;
+            extendsTypes = apiClass.extendsType ? [apiClass.extendsType] : [];
+            break;
+          }
+          case ApiItemKind.Interface: {
+            const apiInterface: ApiInterface = next as ApiInterface;
+            extendsTypes = apiInterface.extendsTypes;
+            break;
+          }
+        }
+
+        if (extendsTypes === undefined) {
+          messages.push({
+            messageId: FindApiItemsMessageId.UnsupportedKind,
+            text: `Item ${next.displayName} is of unsupported kind ${next.kind}.`
+          });
+          maybeIncompleteResult = true;
+          break;
+        }
+
+        for (const extendsType of extendsTypes) {
+          // We want to find the reference token associated with the actual inherited declaration.
+          // In every case we support, this is the first reference token. For example:
+          //
+          // ```
+          // export class A extends B {}
+          //                        ^
+          // export class A extends B<C> {}
+          //                        ^
+          // export class A extends B.C {}
+          //                        ^^^
+          // ```
+          const firstReferenceToken: ExcerptToken | undefined = extendsType.excerpt.spannedTokens.find(
+            (token: ExcerptToken) => {
+              return token.kind === ExcerptTokenKind.Reference && token.canonicalReference;
+            }
+          );
+
+          if (!firstReferenceToken) {
+            messages.push({
+              messageId: FindApiItemsMessageId.UnexpectedExcerptTokens,
+              text: `Encountered unexpected excerpt tokens in ${next.displayName}. Excerpt: ${extendsType.excerpt.text}.`
```

Comment:
`!firstReferenceToken` occurs if the excerpt does not contain any token with a `canonicalReference`.  The analyzer can generate canonical references for a very wide variety of declarations, including things that don't get an `ApiItem`, so it's interesting to consider how this situation might arise.  Here's one example:

```ts
/** @public */
export declare class MyClass extends (class { }) { 
}
```

We can imagine other edge cases that should have references, but [DeclarationReferenceGenerator](https://github.com/microsoft/rushstack/blob/15680b899a06c614c32088b02940bf676596bcf2/apps/api-extractor/src/generators/DeclarationReferenceGenerator.ts#L17) hasn't implemented support (maybe an augmented type or somesuch).

Is "Encountered unexpected excerpt tokens" the best way to describe these cases?  Here's a couple alternative ideas:

- Safe approach: "A canonical reference was not found in the extends clause" accurately reports the implementation logic that failed
- Friendly approach: "The extends clause does not refer to type that can be analyzed" tries to provide a meaningful interpretation of the situation that will hopefully be correct